WebHostLib/customserver.py

```diff
@@ -137,8 +139,45 @@ def get_save(self) -> dict:
         return d
 
 
-def get_random_port():
-    return random.randint(49152, 65535)
+def get_random_port(game_ports):
+    config_range_list = game_ports.split(",")
+    available_ports = []
+    for item in config_range_list:
+        if '-' in item:
+            start, end = map(int, item.split('-'))
+            available_ports.extend(range(start, end+1))
+        else:
+            available_ports.append(int(item))
+
+    port = get_port_from_list(available_ports)
+    if port == 0:
+        checked_ports = []
+        while len(set(checked_ports)) < (65535-49152)+1:
+            port = random.randint(49152, 65535)
+            if not is_port_in_use(port):
+                break
+            else:
+                checked_ports.append(port)
+        logging.info(f"Unable to find an available port in custom range. Expanded search to the default ports. Hosting on port {port}.")
+
+    return port
+
+
+def get_port_from_list(available_ports: list) -> int:
+    while available_ports:
+        port = random.choice(available_ports)
+        available_ports.remove(port)
+
+        if not is_port_in_use(port):
```

Comment:
This is racy. Instead the upper layer should handle the exception when it tries to listen.